compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -777,6 +777,9 @@ impl CodegenCx<'b, 'tcx> {
 
         // This isn't an "LLVM intrinsic", but LLVM's optimization passes
         // recognize it like one and we assume it exists in `core::slice::cmp`
+        #[cfg(c_int_width = "16")]
+        ifn!("memcmp", fn(i8p, i8p, t_isize) -> t_i16);
+        #[cfg(c_int_width = "32")]
         ifn!("memcmp", fn(i8p, i8p, t_isize) -> t_i32);
```

Comment:
I don't think this is right.  This code is part of the compiler and as such your `cfg()`s apply to the build architecture, not the one you're compiling for.  I think this needs to be something like

```rust
match self.sess().target.c_int_width  {
    16 => ifn!("memcmp", fn(i8p, i8p, t_isize) -> t_i16),
    32 => ifn!("memcmp", fn(i8p, i8p, t_isize) -> t_i32),
    _ => /* what to do as the fallback? */,
}
```

Reply:
Also, there exist platforms with a 64-bit c_int (ILP64 platforms), though I don't think we currently support any.

Reply:
Oh!  Yes.  This makes perfect sense.  Thank you!  I'll see if I can fix this up today.